compiler/noirc_evaluator/src/ssa/ir/instruction/cast.rs

```diff
@@ -27,10 +27,10 @@ pub(super) fn simplify_cast(
                 SimplifiedTo(value)
             }
             (
-                Type::Numeric(NumericType::Unsigned { .. }),
+                Type::Numeric(NumericType::Unsigned { .. } | NumericType::Signed { .. }),
                 Type::Numeric(NumericType::NativeField),
             ) => {
-                // Unsigned -> Field: redefine same constant as Field
```

Comment:
This got me thinking that it's a bit strange we convert `-1` to `2**N - 1` rather than to `p-1`. It's preexisting behaviour and would need discussion so it's not a blocker for this PR.